### PR TITLE
Migrate from vCard 2.1 to vCard 3.0 with utf-8

### PR DESCRIPTION
**Pull request by schast**
- [x] migrate format to vCard 3.0 to support import in modern systems like nextcloud
- [x] switch from iso-8859-1 to utf-8
- [x] only export existing fields
- [x] add REV and UID field to support vcard merging (do not import multiple)

### CODE REVIEW

File: adm_program/modules/groups-roles/groups_roles_function.php [hunk 1]
```diff
@@ -290,7 +290,7 @@
 
         $filename = FileSystemUtils::getSanitizedPathEntry($filename);
 
-        header('Content-Type: text/x-vcard; charset=iso-8859-1');
```
Comment (Fasse) [inline adm_program/modules/groups-roles/groups_roles_function.php:293]: As I understand https://de.wikipedia.org/wiki/VCard the content-type text/x-vcard should be still used for vcard version 3.0

File: adm_program/system/classes/User.php [hunk 2]
```diff
@@ -1119,93 +1119,109 @@ public function getValue(string $columnName, string $format = '')
...
             }
-
-            $vCard[] = 'X-GENDER:' . $xGender;
-            $vCard[] = 'X-WAB-GENDER:' . $xWabGender;
         }
         if ($this->getValue('usr_timestamp_change') !== '') {
-            $vCard[] = 'REV:' . $this->getValue('usr_timestamp_change', 'Ymd\This');
+            $vCard[] = 'REV:' . $this->getValue('usr_timestamp_change', 'c');
+        }
+        if ($this->getValue('usr_uuid') !== '') {
```
Comment (Fasse) [inline adm_program/system/classes/User.php:1221]: Every user record should have a usr_uuid so you don't need to check if it's empty or not.

**Review by Fasse — COMMENTED** [review]
Nice enhancement!

**Review by Fasse — CHANGES_REQUESTED** [review]
Please view my comments.

File: adm_program/system/classes/User.php [hunk 3]
```diff
@@ -1119,93 +1119,109 @@ public function getValue(string $columnName, string $format = '')
...
+            } elseif ((int)$this->getValue('GENDER') === 2) {
+                $vCard[] = 'GENDER:F';
+            } elseif ((int)$this->getValue('GENDER') === 3) {
+                $vCard[] = 'GENDER:O';
             }
-
-            $vCard[] = 'X-GENDER:' . $xGender;
-            $vCard[] = 'X-WAB-GENDER:' . $xWabGender;
         }
         if ($this->getValue('usr_timestamp_change') !== '') {
```
Comment (Fasse) [inline adm_program/system/classes/User.php:1218]: I don't know exactly how REV worked. But within iCal there is a similar field and that must always be filled. First with usr_timestamp_create and if set than with usr_timestamp_change.

File: adm_program/system/classes/User.php (hunk 2 above)
Comment (schast) [reply adm_program/system/classes/User.php:1221]: thanks, I will fix that

File: adm_program/system/classes/User.php (hunk 3 above)
Comment (schast) [reply adm_program/system/classes/User.php:1218]: ok, REV field is now set always.

**Review by Fasse — APPROVED** [review]

**Pull request merged**